### ISSUE TITLE
Slow ContextMenu

### ISSUE THREAD

**Issue by seberoth** [WolvenKit/WolvenKit#2006]
Calls to `ModifierViewStateService.RefreshModifierStates()` slows down every ContextMenu using it, because it fires an event (`ModifierViewStateService.ModifierStateChanged`) to ALL subscribers even if they aren't needed for the current action.

I.e. Opening the ContextMenu in the ProjectExplorer gets slowed down while a massive CR2W file is open.

**Comment by seberoth** [comment]
Also happens when you try to paste something into the AssetBrowser search bar. 
Took 45 seconds until CVM was finished to recalculate everything...

**Comment by poirierlouis** [comment]
I wonder if ContextMenu is the only UI part affected?
imo, there is a lot going on regarding slowness and UI freeze throughout the app.
It is something I'd like to look at after finishing #1954 . Thanks for the clarification, this is one step among many to come...

**Comment by manavortex** [comment]
Yeah, it's slow and I don't like it. Sorry for that (since I messed it up).
I noticed that especially NotifyCanExecuteChanged is super expensive in regards to input lag. I'd be grateful if you could fix this.
